### PR TITLE
Add support for additional OpenCL functions

### PR DESCRIPTION
**Pull request by rok-cesnovar**
This adds support for additional OpenCL supported functions, those that do not return scalar. 

For now we will deal with them a bit naively as

`from_matrix_cl(foo(to_matrix_cl(arg1), to_matrix_cl(arg2))`

## Release notes

Added support for additional OpenCL functions:  beta, crossprod, cholesky_decompose, digamma, inv_cloglog, inv_Phi, lbeta, lgamma, log1m_exp, log1m_inv_logit, log1p_exp, log_diff_exp, log_inv_logit, log_inv_logit_diff, mdivide_left_tri_low, mdivide_right_tri_low, multiply, rows_dot_product, rows_dot_self, tcrossprod, tgamma

## Copyright and Licensing

By submitting this pull request, the copyright holder is agreeing to 
license the submitted work under the BSD 3-clause license (https://opensource.org/licenses/BSD-3-Clause)

### CODE REVIEW

**Comment by rok-cesnovar** [comment]
@seantalts this one is a bit last minute, but is really simple. Do you maybe have time to look at this?

File: src/stan_math_backend/Transform_Mir.ml [hunk 1]
```diff
@@ -46,10 +46,18 @@ let opencl_supported_functions =
   ; "poisson_lpmf"; "poisson_log_lpmf"; "poisson_log_glm_lpmf"; "rayleigh_lpdf"
   ; "scaled_inv_chi_square_lpdf"; "skew_normal_lpdf"; "std_normal_lpdf"
   ; "student_t_lpdf"; "uniform_lpdf"; "weibull_lpdf" ]
+  @ [ "beta"; "crossprod"; "cholesky_decompose"; "digamma"; "inv_cloglog"
+    ; "inv_Phi"; "lbeta"; "lgamma"; "log1m_exp"; "log1m_inv_logit"; "log1p_exp"
+    ; "log_diff_exp"; "log_inv_logit"; "log_inv_logit_diff"
+    ; "mdivide_left_tri_low"; "mdivide_right_tri_low"; "multiply"
+    ; "rows_dot_product"; "rows_dot_self"; "tcrossprod"; "tgamma" ]
```
Comment (rok-cesnovar) [inline src/stan_math_backend/Transform_Mir.ml:53]: The list of supported functions is longer (see Supported functions
 in http://mc-stan.org/math/opencl_support.html) but these are typically faster standalone (with data transfers) as shown by Math benchmarks.

**Comment by rok-cesnovar** [comment]
Actually, never mind. Lets wait for `var<Eigen::Matrix>` support and a more advanced OpenCL transform and to make this more optimized.

**Comment by tpfau** [comment]
Is there any plan for these functions to be added at some point? 
We currently have a model which needs to do a quite large cholesky decompose and pushing that over to the GPU might help speeding things up.

**Comment by rok-cesnovar** [comment]
Hi, 

yeah, there is a plan to expose all of them on the Stan level. Most functions have GPU support in the backend but have not been exposed at the language level because it is a bit tricky to do it reliably. I hope we get to this in this release cycle.

**Comment by tpfau** [comment]
Sounds great :+1:

**Comment by WardBrian** [comment]
@rok-cesnovar what is the blocker for this, the `@` annotations?

**Comment by rok-cesnovar** [comment]
There are two things we could do: 
a) piggyback off of what we have for varmat/SoA. Meaning that if `--O1` (or higher) and `--use-opencl` move everything that is marked `SoA` to the GPU equivalent. varmat supported functions are a subset of OpenCL supported functions, so this one should be fairly easy in terms of what we have to do in stanc3. Would really love to do this one as I think it's a low-hanging fruit.

b) annotations to use GPUs more broadly than just what is caught with a)

**Pull request closed without merging**